subprojects/dependency-management/src/integTest/groovy/org/gradle/integtests/resolve/forsubgraph/InheritConstraintsInPlatformCentricDevelopmentIntegrationTest.groovy

```diff
@@ -0,0 +1,430 @@
+        when:
+        repositoryInteractions {
+            'org:platform' {
+                expectVersionListing()
+            }
+            'org:platform:1.1' {
+                expectGetMetadata()
+                if (platformType == MODULE) { expectGetArtifact() }
+            }
+            'org:bar:2.0' {
+                expectGetMetadata()
+                expectGetArtifact()
+            }
+            'org:foo:3.1.1' {
+                expectGetMetadata()
+                expectGetArtifact()
+            }
+        }
+        run ':checkDeps'
+
+        then:
+        resolve.expectGraph {
+            root(':', ':test:') {
+                edge('org:platform:1.+', 'org:platform:1.1') {
+                    byRequest()
+                    if (platformType != MODULE) {
+                        configuration(platformType == ENFORCED_PLATFORM ? 'enforcedApiElements' : 'apiElements')
+                        noArtifacts()
+                    }
+                    constraint("org:bar:${expectSubgraphVersion(platformType, '2.0')}", 'org:bar:2.0').byConstraint()
+                    constraint("org:foo:${expectSubgraphVersion(platformType, '3.1.1', '3.1 & 3.2')}", 'org:foo:3.1.1').byConstraint()
+                }
+                edge('org:bar', 'org:bar:2.0') {
+                    byRequest()
+                    edge('org:foo:3.1', 'org:foo:3.1.1') {
+                        if (platformType != ENFORCED_PLATFORM) { byAncestor() } else { byRequest() }
+                    }
+                }
+            }
+            if (platformType == ENFORCED_PLATFORM) {
+                nodesWithoutRoot.each { it.forced() }
+            }
+        }
+
+        where:
+        platformType << PlatformType.values()
+    }
+
+    @Unroll
+    void "(3) library developer has issues with org:foo:3.1.1 and overrides platform decision with 3.2 which fails due to reject [#platformType]"() {
+        updatedRepository(platformType)
+        singleLibraryBuildFile(platformType)
+        buildFile << """
+            dependencies {
+                constraints {
+                    conf('org:foo:3.2')
+                }
+            }
+        """
+
+        when:
+        repositoryInteractions {
+            'org:platform' {
+                expectVersionListing()
+            }
+            'org:platform:1.1' {
+                expectGetMetadata()
+            }
+            'org:bar:2.0' {
+                expectGetMetadata()
+                if (platformType == ENFORCED_PLATFORM) { expectGetArtifact() }
+            }
+            'org:foo:3.1.1' {
+                expectGetMetadata()
+                if (platformType == ENFORCED_PLATFORM) { expectGetArtifact() }
+            }
+            'org:foo:3.2' {
+                if (platformType != ENFORCED_PLATFORM) { expectGetMetadata() }
+            }
+        }
+        if (platformType == ENFORCED_PLATFORM) {
+            // issue with enforced platform: consumer can not override platform decision
```

Comment:
Technically you can, with a first level `force`

Reply:
Ok I have not tried. I'll adjust one of the tests to demonstrate this.

Reply:
I changed the comments in the test.